config.go

```diff
@@ -187,11 +195,17 @@ type cacheKey struct {
 }
 
 func (c *frozenConfig) addDecoderToCache(fingerprint [32]byte, rtype uintptr, dec ValDecoder) {
+	if c.config.DisableDecoderCaching {
```

Comment:
All these cases need tests please.

Reply:
I actually tried looking at config_internal_test.go but as far as I know, the tests currently in place do not validate anything. Regardless of the config options, the resulting interface will be the same so doing an assert.Equals() does not validate whether or not the decoder/encoder is fetched from cache or not.

I tried changing to assert.Same() to compare the memory address of both pointers but that did not work either. Stepping in the code, I validated it was behaving as expected but could not do more. Do you have any ideas? I wanted to change the function return values to include a boolean indicating if we got a cache hit or not but that would break the interface API so thats also a no-go.

Reply:
The tests are fixed in #218

Reply:
Added coverage